include/zephyr/sys/device_mmio.h

```diff
@@ -101,8 +105,11 @@ static inline void device_map(mm_reg_t *virt_addr, uintptr_t phys_addr,
 #else
 	ARG_UNUSED(size);
 	ARG_UNUSED(flags);
-
+#ifdef CONFIG_EXTERNAL_ADDRESS_TRANSLATION
+	sys_mm_drv_page_phys_get((void *) phys_addr, virt_addr);
```

Comment:
I have no objection to this since this is the only use case for now?

Just raise my concern for the future as this is a system API:
1. MMU/MPU and RAT (or alternative technologies) exist at the same time which is hard to handle.
2. Suppose we remove the exclusive macro (like `z_phys_map` firstly, and then `sys_mm_drv_page_phys_get`, it's still annoying that if we want to remap RAT only, however, we inevitably call MMU remap again.
3.  How do we deal with similar techniques like Arm ATU?

Fortunately, we could think about it later. MMU and RAT existing simultaneously seem like a 2 stage translation which is definitely a tricky thing.